### PR TITLE
Moredetail activity UI & Image gallery view popup

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 75703: More Detail screen ](https://edu-redmine.sun-asterisk.vn/issues/75703)
## WHAT
- Moredetail activity UI & Image gallery view popup
## Evidence (Screenshot or Video)

https://github.com/awesome-academy/HN_Android_2024_Travenor_App/assets/165371062/12cafe78-7d40-416d-b2e3-bac6ce93c893

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        val remoteDataSource = PlaceRemoteDataSource.getInstance()
+        val localDataSource = PlaceLocalDataSource.getInstance(baseContext)
+        val localExploreDataSource = PlaceExploreLocalSource.getInstance(baseContext)
+
+        val placeRepositoryImpl = PlaceRepositoryImpl.getInstance(
+            remoteDataSource,
+            localDataSource,
+            localExploreDataSource
+        )
+
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt:50]: mấy cái này k thay đổi cho lên khai báo lazy cho gọn cái initView cũng đc

File: app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    }
+
+    override fun onGetPlaceDetailFail(e: Exception) {
+        finish()
+    }
+
+    private fun openPlaceImageGalleryView(position: Int) {
+        val imageViewDialogFragment = ImageViewDialogFragment.newInstance(photos, position)
+        imageViewDialogFragment.show(supportFragmentManager, "ImageViewDialogFragment")
+    }
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt:94]: tạo cái tag bên class đấy

File: app/src/main/java/com/example/travenor/screen/popup/imageview/ImageViewDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        imageContent = dialog?.findViewById(R.id.image_content)
+        loadImageContent(selectedPosition)
+
+        val recyclerView = dialog?.findViewById<RecyclerView>(R.id.recyclerView)
+        recyclerView?.adapter = this.adapter
+
+        dialog?.findViewById<ImageButton>(R.id.button_dismiss)
+            ?.setOnClickListener { _ -> dialog?.dismiss() }
+
+        return dialog!!
```
Comment (ghost) [inline app/src/main/java/com/example/travenor/screen/popup/imageview/ImageViewDialogFragment.kt]: cái này bên trên check null nma bên dưới !! a thấy k hợp lí lắm nhỉ

Comment (ghost) [reply app/src/main/java/com/example/travenor/screen/popup/imageview/ImageViewDialogFragment.kt]: Em sẽ đổi sang requrireContext để không gây null + sử dụng with

File: app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt (hunk 2 above)
Comment (ghost) [reply app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt:94]: ok a

File: app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt (hunk 1 above)
Comment (ghost) [reply app/src/main/java/com/example/travenor/screen/moredetail/MoreDetailActivity.kt:50]: local variable, chỉ dùng 1 lần nên e k khai báo global

**Review by ghost — APPROVED** [review]

**Review by daolq-2712 — APPROVED** [review]
LGTM!

**Pull request merged**